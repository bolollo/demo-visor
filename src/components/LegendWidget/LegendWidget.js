// @flow

import React, { useState } from "react";
import PropTypes from "prop-types";
import { Icon } from "semantic-ui-react";

import "semantic-ui-css/components/icon.min.css";
import styles from "./LegendWidget.module.css";

export default function LegendWidget({position, ranges, colors, showInitial}) {

	const [show, setShow] = useState(showInitial);

	const positionClassName = styles[position];

	const toggleLegend = () => {

		setShow(!show);

	};

	return (
		<div className={[styles.legendWidget, positionClassName, show ? "" : styles.hidden].join(" ")}>
			<div className={styles.icon} onClick={toggleLegend}>
				{show ? <Icon link name="sort down" /> : <Icon link name="sort up" />}
			</div>
			{show && ranges.map((item, index) => (
				<div key={index}>
					<div className={styles.legendBlock} style={{backgroundColor: colors[index]}}></div>
					<div>{item}</div>
				</div>
			)
			)}
		</div>
	);

}

LegendWidget.propTypes = {
	position: PropTypes.oneOf(["bottom-right", "bottom-left", "top-right", "top-left", "relative"]),
	ranges: PropTypes.array,
	colors: PropTypes.array,
	showInitial: PropTypes.bool
};

LegendWidget.defaultProps = {
	position: "bottom-right",
	ranges: [],
	colors: [],
	showInitial: false,
};
