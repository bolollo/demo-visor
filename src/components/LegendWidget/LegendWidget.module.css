.legendWidget{
    z-index:100;
    margin: 0.5rem;
    padding: 5px;
    background-color: var(--primary-white, #ffffff);
    border-radius: 5px;
    min-width: 150px;
    opacity: 0.95;
    clear:left;
    font-size: 0.8em;
}

.icon {
    cursor: pointer;
    text-align: right;
}

.hidden {
    min-width: 24px;
}

.legendBlock {
    border: 1px solid;
    border-color: var(--dark-grey, #555555);
    display: block;
    float: left;
    height: 12px;
    margin: 0 5px 0 0px;
    width: 20px;
}

.bottom-right{
    position: absolute;
    bottom: 0;
    right: 0;
}
.bottom-left{
    position: absolute;
    bottom: 0;
    left: 0;
}
.top-right{
    position: absolute;
    top: 0;
    right: 0;
}
.top-left{
    position: absolute;
    top: 0;
    left: 0;
}
.relative{
    position: relative;
}