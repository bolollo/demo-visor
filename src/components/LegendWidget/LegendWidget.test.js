// @flow

import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import LegendWidget from "./LegendWidget";

const ranges = [
	"1 - 2",
	"3 - 4",
	"4 - 5"
];

const colors = [
	"#ff0000",
	"#00ff00",
	"#0000ff"
];

describe("LegendWidget component", () => {

	it("matches the snapshot ", () => {

		const component = shallow(<LegendWidget></LegendWidget>);
		expect(toJSON(component)).toMatchSnapshot();

	});

	it("showInitial has 3 div", () => {

		const component = shallow(<LegendWidget showInitial={true} ranges={ranges} colors={colors}></LegendWidget>);
		expect(component.find(".legendBlock")).toHaveLength(3);

	});

	it("has position bottom-left", () => {

		const component = shallow(<LegendWidget position={"bottom-left"}></LegendWidget>);
		expect(component.find(".legendWidget").hasClass("bottom-left")).toEqual(true);

	});

});
