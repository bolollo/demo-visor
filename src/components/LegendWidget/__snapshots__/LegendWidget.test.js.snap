// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LegendWidget component matches the snapshot  1`] = `
<div
  className="legendWidget bottom-right hidden"
>
  <div
    className="icon"
    onClick={[Function]}
  >
    <Icon
      as="i"
      link={true}
      name="sort up"
    />
  </div>
</div>
`;
